### PR TITLE
Feature: improve serialization of email string

### PR DESCRIPTION
**Pull request by oloko64**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,103 +1,163 @@
+use serde::Serialize;
 use std::error::Error;
 
+#[derive(Clone)]
```
Comment (ArturVRSampaio) [inline src/lib.rs]: DESNECESSAURO

File: src/lib.rs [hunk 2]
```diff
@@ -1,103 +1,163 @@
...
     ///    Err(err) => println!("Error sending email: {}", err),
     /// }
     /// ```
-    pub fn send(&self, subject: &str, body: &str) -> Result<(), Box<dyn Error>> {
+    pub fn send<T: AsRef<str>>(&mut self, subject: T, body: T) -> Result<(), Box<dyn Error>> {
         ureq::post("https://api.sendgrid.com/v3/mail/send")
             .set("Authorization", &format!("Bearer {}", self.api_key))
             .set("Content-Type", "application/json")
-            .send_string(&self.set_body_email(subject, body))?;
+            .send_string(&self.set_body_email(subject.as_ref(), body.as_ref()))?;
```
Comment (ArturVRSampaio) [inline src/lib.rs:100]: substituir refs por &

File: src/lib.rs [hunk 3]
```diff
@@ -1,103 +1,163 @@
...
+    }
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq)]
+pub struct SendgridEmail<'a> {
+    #[serde(skip)]
+    api_key: String,
+
+    #[serde(rename = "personalizations")]
+    personalizations: Vec<Personalization>,
```
Comment (ArturVRSampaio) [inline src/lib.rs]: verificar as possiveis personalizations... se nao faz sentido usar todos, criar variavel personalization que contem somente a usada

Comment (ArturVRSampaio) [reply src/lib.rs]: personalization: personalizations[0]

Comment (ArturVRSampaio) [reply src/lib.rs]: personalization: Vec<Personalization>.first

File: src/lib.rs [hunk 4]
```diff
@@ -1,103 +1,163 @@
...
     /// ```
     pub fn add_cc_emails(&mut self, cc_emails: &[&'a str]) {
-        match self.cc_emails.as_mut() {
-            Some(cc) => cc.extend(cc_emails),
-            None => self.cc_emails = Some(cc_emails.to_vec()),
+        match self.personalizations[0].cc.as_mut() {
+            Some(cc) => {
+                for email in cc_emails {
+                    cc.push(From {
+                        email: (*email).to_string(),
```
Comment (ArturVRSampaio) [inline src/lib.rs]: var = &var1
*var
T_T

**Review by ArturVRSampaio — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (oloko64) [reply src/lib.rs:100]: This variable is a generic, it needs to be known as a `str`

**Pull request merged**